lang/cs/Org.Apache.REEF.IMRU/API/UpdateResult.cs

```diff
@@ -0,0 +1,87 @@
+﻿namespace Org.Apache.REEF.IMRU.API
+{
+    /// <summary>
+    /// Represents the output of an IUpdateFunction.
+    /// </summary>
+    /// <seealso cref="IUpdateFunction{TMapInput,TMapOutput,TResult}" />
+    /// <typeparam name="TMapInput"></typeparam>
+    /// <typeparam name="TResult"></typeparam>
+    public sealed class UpdateResult<TMapInput, TResult>
+    {
+        private readonly bool _done;
+        private readonly bool _hasMapInput;
+        private readonly bool _hasResult;
+        private readonly TMapInput _mapInput;
+        private readonly TResult _result;
+
+        private UpdateResult(bool done, bool hasMapInput, bool hasResult, TResult result, TMapInput mapInput)
+        {
+            _result = result;
```

Comment:
Yes, but until C# 6 arrives, those properties wouldn't be read-only. Instead, they'd have to have a `private set` method. I prefer not to have that as it makes reasoning about the thread safety of classes easier.